lib/spack/docs/spack_workflows.rst

```diff
@@ -0,0 +1,583 @@
+===============
+Spack Workflows
+===============
+
+The process of using Spack involves building packages, running
+binaries from those packages, and developing software that depends on
+those packages.  For example, one might use Spack to build the
+`netcdf` package, use `spack load` to run the `ncdump` binary, and
+finally, write a small C program to read/write a particular NetCDF file.
+
+Spack supports a variety of workflows to suit a variety of situaions
+and user preferences, there is no single way to do all these things.
+This chapter demonstrates different workflows that have been
+developed, pointing out the pros and cons of them.
+
+###########
+Definitions
+###########
+
+First some basic definitions.
+
+-----------------------------------------
+Package, Concrete Spec, Installed Package
+-----------------------------------------
+
+In Spack, a package is an abstract recipe to build one piece of software.
+Spack packages may be used to build, in principle, any version of that
+software with any set of variants.  Examples of packages include
+``curl`` and ``zlib``.
+
+A package may be *instantiated* to produce a concrete spec; one
+possible realization of a particular package, out of combinatorially
+many other realizations.  For example, here is a concrete spec
+instantiated from ``curl``:
+
+.. code-block:: sh
```

Comment:
This code-block should probably be formatted as "none". It's not shell code, and Sphinx may complain that it can't figure out how to highlight it.